lightning/src/ln/channelmanager.rs

```diff
@@ -3970,6 +4005,12 @@ where
 								node_id: channel.get_counterparty_node_id(),
 								msg: announcement_sigs,
 							});
+						} else if channel.is_usable() {
+							log_trace!(self.logger, "Sending funding_locked WITHOUT announcement_signatures but with private channel_update for our counterparty on channel {}", log_bytes!(channel.channel_id()));
+							pending_msg_events.push(events::MessageSendEvent::SendChannelUpdate {
+								node_id: channel.get_counterparty_node_id(),
+								msg: self.get_channel_update_for_unicast(channel).unwrap(),
+							});
```

Comment:
Don't quite get why we don't push a `SendChannelUpdate` event when `announcement_sigs.is_some()`?

Reply:
If we send a `channel_update` which we intend to have relayed, we have to send a `channel_announcement` first. If we're just sending it to our counterparty they shouldn't care, but if the channel is public we have to wait until announcement_signatures are exchanged so we can send a channel_announcement.